lightningd/channel_control.c

```diff
@@ -303,6 +303,10 @@ void peer_start_channeld(struct channel *channel,
 	const struct config *cfg = &ld->config;
 	bool reached_announce_depth;
 	struct secret last_remote_per_commit_secret;
+	secp256k1_ecdsa_signature *remote_ann_node_sig, *remote_ann_bitcoin_sig;
+
+	remote_ann_node_sig = tal(tmpctx, secp256k1_ecdsa_signature);
+	remote_ann_bitcoin_sig = tal(tmpctx, secp256k1_ecdsa_signature);
```

Comment:
This is the weird ownership I was talking about before. Please have `wallet_remote_ann_sigs_load` manage allocations. The only reason we aren't producing a memory leak here is that `tmpctx` and all child allocations is cleared in the `io_loop`.

Reply:
Thank you so much! Now I pass `ctx `to `wallet_remote_ann_sigs_load` and let it manage allocations.